children/schema.py

```diff
@@ -0,0 +1,95 @@
+import graphene
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from graphene_django.types import DjangoObjectType
+
+from users.models import Guardian
+
+from .models import Child, Relationship
+
+User = get_user_model()
+
+
+class ChildType(DjangoObjectType):
+    class Meta:
+        model = Child
+
+
+class ChildInput(graphene.InputObjectType):
+    first_name = graphene.String(required=True)
+    last_name = graphene.String(required=True)
+    birthdate = graphene.Date(required=True)
+
+
+# this class cannot be named just RelationshipType because of graphene-django
+# https://github.com/graphql-python/graphene-django/issues/185
+class RelationshipObjectType(DjangoObjectType):
+    class Meta:
+        model = Relationship
+        fields = ("type", "child", "guardian")
+
+
+class RelationshipTypeEnum(graphene.Enum):
+    PARENT = "parent"
+    OTHER_GUARDIAN = "other_guardian"
+    OTHER_RELATION = "other_relation"
+    ADVOCATE = "advocate"
+
+
+class RelationshipInput(graphene.InputObjectType):
+    type = RelationshipTypeEnum()
+
+
+class GuardianType(DjangoObjectType):
+    class Meta:
+        model = Guardian
+
+
+class GuardianInput(graphene.InputObjectType):
+    first_name = graphene.String(required=True)
+    last_name = graphene.String(required=True)
+    email = graphene.String(required=True)
+    phone = graphene.String()
+
+
+class SubmitChildMutation(graphene.relay.ClientIDMutation):
+    class Input:
+        child = ChildInput(required=True)
+        relationship = RelationshipInput()
+        guardian = GuardianInput(required=True)
+
+    child = graphene.Field(ChildType)
+    relationship = graphene.Field(RelationshipObjectType)
+    guardian = graphene.Field(GuardianType)
+
+    @classmethod
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO we'll probably need to check somehow if the child already exists
+        child = Child.objects.create(**kwargs["child"])
+
+        # TODO change this to logged in user once we have authentication in place
+        user_obj, _ = User.objects.get_or_create(username="dummyman777")
```

Comment:
I guess this is good for the time being, until the logging in is implemented. Might as well do just `.create()` maybe, since you create the `Child` on each mutation anyway.

Reply:
True, that could actually be better.